cmake/XrplSettings.cmake

```diff
@@ -62,7 +58,14 @@ else()
   set(wextra OFF CACHE BOOL "gcc/clang only" FORCE)
 endif()
 
-if(is_linux)
+# Sanitizer configuration from environment (flags set in XrplSanitizers.cmake)
+if($ENV{SANITIZERS})
+  set(SANITIZER TRUE)
+else()
+  set(SANITIZER FALSE)
+endif()
```

Comment:
Let's move this to `CompilationEnv`, and only use the variable `SANITIZERS` (a string value)